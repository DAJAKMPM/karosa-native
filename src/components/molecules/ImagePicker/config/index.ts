/**
 *
 * @format
 *
 */

export const ASPECT_RATIO: [number, number] = [4, 3];
export const QUALITY = 1;

export const SIZE = {
  PRIMARY: 28,
  SECONDARY: 24,
};
