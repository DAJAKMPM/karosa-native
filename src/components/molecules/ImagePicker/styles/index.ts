/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { DIMENS, theme } from "@app/styles";

const ImagePickerStyles = StyleSheet.create({
  variationOneContainer: {
    alignItems: "center",
    backgroundColor: theme.colors.light15,
    borderRadius: 5,
    borderColor: theme.colors.dark10,
    height: DIMENS.screenHeight * 0.12,
    justifyContent: "center",
    overflow: "hidden",
    width: DIMENS.screenWidth * 0.22,
  },
  variationTwoContainer: {
    alignItems: "center",
    backgroundColor: theme.colors.dark35,
    borderWidth: 4,
    borderStyle: "solid",
    borderColor: theme.colors.white,
    borderRadius: 50,
    height: DIMENS.screenHeight * 0.14,
    justifyContent: "center",
    overflow: "hidden",
    width: DIMENS.screenWidth * 0.24,
  },
  variationThreeContainer: {
    alignItems: "center",
    backgroundColor: theme.colors.dark35,
    height: DIMENS.screenHeight * 0.23,
    justifyContent: "center",
    overflow: "hidden",
    width: DIMENS.screenWidth,
  },
  variationFourMainContainer: {
    width: 82,
    height: 76,
    borderWidth: 0.5,
    borderStyle: "solid",
    borderColor: theme.colors.primary,
    borderRadius: 5,
    overflow: "hidden",
    position: "relative",
    justifyContent: "center",
  },
  addIcon: {
    position: "absolute",
    zIndex: 99,
    alignSelf: "center",
  },
  image: {
    width: DIMENS.screenWidth,
    height: "100%",
    opacity: 0.5,
    zIndex: 21,
  },
  variationFourCamera: {
    width: "100%",
    height: "100%",
    borderRadius: 5,
    resizeMode: "cover",
    opacity: 0.5,
  },
  variationFourSubContainer: {
    width: 82,
    height: 76,
    borderWidth: 0.5,
    borderStyle: "solid",
    borderColor: theme.colors.primary,
    borderRadius: 5,
  },
  iconContainer: {
    flexGrow: 1,
    padding: 16,
    alignItems: "center",
    justifyContent: "center",
  },
  cameraIcon: {
    zIndex: 1,
    position: "absolute",
  },
  txtAddImg: {
    ...theme.textSmall,
    textAlign: "center",
    fontWeight: "300",
    color: theme.colors.black,
  },
  txtContainer: {
    paddingHorizontal: 8,
    paddingVertical: 2,
    backgroundColor: theme.colors.primary,
    borderBottomLeftRadius: 5,
    borderBottomRightRadius: 5,
  },
  txtVariationFour: {
    color: theme.colors.white,
    ...theme.textLight,
  },
});

export default ImagePickerStyles;
