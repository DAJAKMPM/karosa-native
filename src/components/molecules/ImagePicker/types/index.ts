/**
 *
 * @format
 *
 */

export type PropsType = {
  variation: number;
  uri: string | null;
  onChange(img: string | null): void;
};

export type VariationType = {
  uri: string | null;
};
