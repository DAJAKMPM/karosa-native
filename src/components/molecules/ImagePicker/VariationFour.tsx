/**
 *
 * VariationFour
 * @format
 *
 */

import React, { FC, Fragment } from "react";
import { View } from "react-native";
import Icon from "@app/atoms/Icon";
import Image from "@app/atoms/Image";
import Text from "@app/atoms/Text";

import type { VariationType } from "./types";
import { SIZE } from "./config";
import ImagePickerStyles from "./styles";

const VariationFour: FC<VariationType> = (props) => {
  const { uri } = props;

  return (
    <Fragment>
      {uri ? (
        <View style={ImagePickerStyles.variationFourMainContainer}>
          <Icon
            group="common"
            name="cameraWhite"
            height={SIZE.PRIMARY}
            width={SIZE.PRIMARY}
            extraStyle={ImagePickerStyles.addIcon}
          />

          <Image
            source={{ uri }}
            imageStyle={ImagePickerStyles.variationFourCamera}
          />
        </View>
      ) : (
        <View style={ImagePickerStyles.variationFourSubContainer}>
          <View style={ImagePickerStyles.iconContainer}>
            <Icon
              group="products"
              name="addImage"
              height={SIZE.TERTIARY}
              width={SIZE.TERTIARY}
            />
          </View>

          <View style={ImagePickerStyles.txtContainer}>
            <Text
              text="Add Image"
              textStyle={ImagePickerStyles.txtVariationFour}
            />
          </View>
        </View>
      )}
    </Fragment>
  );
};

export default VariationFour;
