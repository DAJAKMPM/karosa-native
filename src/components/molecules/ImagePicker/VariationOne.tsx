/**
 *
 * VariationOne
 * @format
 *
 */

import React, { FC, Fragment } from "react";
import { View } from "react-native";
import Icon from "@app/atoms/Icon";
import Image from "@app/atoms/Image";
import Text from "@app/atoms/Text";

import type { VariationType } from "./types";
import { SIZE } from "./config";
import ImagePickerStyle from "./styles";

const VariationOne: FC<VariationType> = (props) => {
  const { uri } = props;

  return (
    <Fragment>
      {uri ? (
        <View style={ImagePickerStyle.variationOneContainer}>
          <Icon
            group="common"
            name="cameraWhite"
            height={SIZE.PRIMARY}
            width={SIZE.PRIMARY}
            extraStyle={ImagePickerStyle.cameraIcon}
          />
          <Image
            source={{ uri }}
            imageStyle={ImagePickerStyle.image}
            resizeMode="contain"
          />
        </View>
      ) : (
        <View style={ImagePickerStyle.variationOneContainer}>
          <Icon
            group="common"
            name="cameraGreen"
            height={SIZE.PRIMARY}
            width={SIZE.PRIMARY}
          />
          <Text text={"Add Photo"} textStyle={ImagePickerStyle.txtAddImg} />
        </View>
      )}
    </Fragment>
  );
};

export default VariationOne;
