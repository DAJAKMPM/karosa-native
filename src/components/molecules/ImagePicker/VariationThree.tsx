/**
 *
 * VariationThree
 * @format
 *
 */

import React, { FC, Fragment } from "react";
import { View } from "react-native";
import Icon from "@app/atoms/Icon";
import Image from "@app/atoms/Image";

import type { VariationType } from "./types";
import { SIZE } from "./config";
import ImagePickerStyle from "./styles";

const VariationThree: FC<VariationType> = (props) => {
  const { uri } = props;

  return (
    <Fragment>
      {uri ? (
        <View style={ImagePickerStyle.variationThreeContainer}>
          <Icon
            group="common"
            name="cameraWhite"
            height={SIZE.SECONDARY}
            width={SIZE.SECONDARY}
            extraStyle={ImagePickerStyle.cameraIcon}
          />
          <Image source={{ uri }} customStyle={ImagePickerStyle.image} />
        </View>
      ) : (
        <View style={ImagePickerStyle.variationThreeContainer}>
          <Icon
            group="common"
            name="cameraGray"
            height={SIZE.SECONDARY}
            width={SIZE.SECONDARY}
          />
        </View>
      )}
    </Fragment>
  );
};

export default VariationThree;
