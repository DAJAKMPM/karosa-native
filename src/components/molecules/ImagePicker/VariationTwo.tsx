/**
 *
 * VariationTwo
 * @format
 *
 */

import React, { FC, Fragment } from "react";
import Icon from "@app/atoms/Icon";
import Image from "@app/atoms/Image";

import { secondarySize } from "./config";
import type { VariationType } from "./types";
import ImagePickerStyle from "./styles";
import { View } from "react-native";

const VariationTwo: FC<VariationType> = (props) => {
  const { uri } = props;

  return (
    <Fragment>
      {uri ? (
        <View style={ImagePickerStyle.variationTwoContainer}>
          <Icon
            group="common"
            name="cameraWhite"
            height={secondarySize}
            width={secondarySize}
            extraStyle={ImagePickerStyle.cameraIcon}
          />
          <Image source={{ uri }} customStyle={ImagePickerStyle.image} />
        </View>
      ) : (
        <View style={ImagePickerStyle.variationTwoContainer}>
          <Icon
            group="common"
            name="cameraGray"
            height={secondarySize}
            width={secondarySize}
          />
        </View>
      )}
    </Fragment>
  );
};

export default VariationTwo;
