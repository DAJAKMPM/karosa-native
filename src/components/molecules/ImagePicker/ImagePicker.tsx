/**
 *
 * ImagePicker
 * @format
 *
 */

import React, { FC, useEffect } from "react";
import { Alert, Platform, View, TouchableWithoutFeedback } from "react-native";
import * as RnImagePicker from "expo-image-picker";
import * as Permissions from "expo-permissions";

import type { PropsType } from "./types";
import { ASPECT_RATIO, QUALITY } from "./config";
import VariationOne from "./VariationOne";
import VariationTwo from "./VariationTwo";
import VariationThree from "./VariationThree";

const ImagePicker: FC<PropsType> = (props) => {
  const { variation, uri, onChange } = props;

  useEffect(() => {
    (async () => {
      if (Platform.OS !== "web") {
        const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
        if (status !== "granted") {
          Alert.alert(
            "Unauthorized Permission",
            "Sorry, we need camera roll permissions to make this work!"
          );
        }
      }
    })();
  }, []);

  const handlePress = () => {
    if (!uri) {
      selectImage();
    } else
      Alert.alert("Delete", "Are you sure you want to delete this image?", [
        { text: "Yes", onPress: () => onChange(null) },
        { text: "No" },
      ]);
  };

  const selectImage = async () => {
    try {
      const result = await RnImagePicker.launchImageLibraryAsync({
        mediaTypes: RnImagePicker.MediaTypeOptions.Images,
        allowsEditing: true,
        aspect: ASPECT_RATIO,
        quality: QUALITY,
      });

      if (!result.cancelled) onChange(result.uri);
    } catch (error) {
      Alert.alert("Error reading an image", error);
    }
  };

  switch (variation) {
    case 1:
      return (
        <TouchableWithoutFeedback onPress={handlePress}>
          <View>
            <VariationOne uri={uri} />
          </View>
        </TouchableWithoutFeedback>
      );
    case 2:
      return (
        <TouchableWithoutFeedback onPress={handlePress}>
          <View>
            <VariationTwo uri={uri} />
          </View>
        </TouchableWithoutFeedback>
      );
    case 3:
      return (
        <TouchableWithoutFeedback onPress={handlePress}>
          <View>
            <VariationThree uri={uri} />
          </View>
        </TouchableWithoutFeedback>
      );
    default:
      return (
        <TouchableWithoutFeedback onPress={handlePress}>
          <View>
            <VariationOne uri={uri} />
          </View>
        </TouchableWithoutFeedback>
      );
  }
};

export default ImagePicker;
