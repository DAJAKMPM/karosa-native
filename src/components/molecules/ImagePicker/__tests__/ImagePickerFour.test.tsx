import React from "react";
import { noop } from "lodash";
import { renderWithWrapper } from "@app/ci/testHelper";
import { COMMON } from "@app/constants";
import ImagePicker from "../ImagePicker";

describe("ImagePicker variation four Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <ImagePicker
        onChange={(_img) => noop()}
        uri=""
        variation={COMMON.VARIATION.FOUR}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
