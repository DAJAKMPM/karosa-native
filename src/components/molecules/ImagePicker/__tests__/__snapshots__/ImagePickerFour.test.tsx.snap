// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePicker variation four Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "borderColor": "#0AA351",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 0.5,
          "height": 76,
          "width": 82,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexGrow": 1,
            "justifyContent": "center",
            "padding": 16,
          }
        }
      >
        <View>
          <SvgMock
            height={24}
            width={24}
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#0AA351",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "paddingHorizontal": 8,
            "paddingVertical": 2,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontFamily": "SFProText-Regular",
              "fontSize": 12,
              "fontWeight": "400",
              "lineHeight": 14.06,
            }
          }
        >
          Add Image
        </Text>
      </View>
    </View>
  </View>
</View>
`;
